src/cpp/server/auth/ServerCSRFToken.cpp

```diff
@@ -44,8 +45,9 @@ void setCSRFTokenCookie(const http::Request& request,
             kCSRFTokenName, 
             csrfToken, 
             "/",  // cookie for root path
-            false // can't be HTTP only since it's read by client script
-            ));
+            false, // can't be HTTP only since it's read by client script
+            // secure if delivered via SSL
+            options().getOverlayOption("ssl-enabled") == "1"));
```

Comment:
There is a call to get overlay options as Boolean which would be more convenient here